src/TuDa.CIMS.Web/Components/AssetItemDialog.razor

```diff
@@ -0,0 +1,239 @@
+﻿@rendermode InteractiveServer
+
+@page "/modal"
+
+@* Modal Button *@
+<button class="btn btn-primary"
+        data-bs-target="#createItemDialog"
+        data-bs-toggle="modal"
+        type="button">
+    Create Item
+</button>
+
+@* Modal Content *@
+<!-- Modal Top -->
+<div aria-hidden="true"
+     aria-labelledby="createItemDialogLabel"
+     class="fade modal"
+     id="createItemDialog"
+     tabindex="-1">
+    <div class="modal-dialog">
+        <div class="modal-content">
+            <div class="modal-header">
+                <h5 class="modal-title"
+                    id="createItemDialogLabel">
+                    Create Item
+                </h5>
+                <button aria-label="Close"
+                        class="btn-close"
+                        data-bs-dismiss="modal"
+                        type="button">
+                </button>
+            </div>
+
+            <!-- Modal Body -->
+            <div class="modal-body">
+
+                <!-- Button Group -->
+                <div aria-label="Button group"
+                     class="btn-group"
+                     role="group">
+                    <button class="btn btn-primary"
+                            @onclick="() => SelectedType = 1">
+                        Chemical
+                    </button>
+                    <button class="btn btn-primary"
+                            @onclick="() => SelectedType = 2">
+                        Consumable
+                    </button>
+                </div>
+
+                <!-- Different Inputs dependent on the Type of the Item -->
+                @if (SelectedType == 1)
```

Comment:
🤔 ♻️ Because of the actual size of the file, maybe we could extract these two into their own component. So there would be a `ConsumableForm` and a `ChemicalForm`. Bad names, but I hope you get the point.

Reply:
Maybe you can also Create a `AssetItem` form and let the other inherit from this so you don't have duplicate code